lib/pbench/server/api/resources/__init__.py

```diff
@@ -37,12 +37,23 @@ class APIAbort(Exception):
     API request.
     """
 
-    def __init__(self, http_status: int, message: Optional[str] = None):
+    def __init__(self, http_status: int, message: Optional[str] = None, **kwargs):
+        """Signal an API failure propagating details through to the base
+        dispatcher's abort call. This allows constraining dependency on the
+        implicit Flask context to the outer layer of the API framework.
+
+        Args
+            http_status: The desired HTTP completion status
+            message: The desired error response message
+            kwargs: Additional error response payload keys/value pairs to
+                communicate additional information.
+        """
         self.http_status = http_status
         self.message = message if message else HTTPStatus(http_status).phrase
+        self.kwargs = kwargs
 
     def __repr__(self) -> str:
-        return f"API error {self.http_status} : {str(self)}"
+        return f"API error {self.http_status} : message={str(self)!r} {self.kwargs}"
```

Comment:
Having the message include the empty braces when `self.kwargs` is empty is slightly disappointing.  How about
```
    def __repr__(self) -> str:
        kw_str = f" {self.kwargs}" if self.kwargs else ""
        return = f"API error {self.http_status} : message={str(self)!r}{kw_str}"
```